tests/ui/unsized-locals/unsized-exprs.rs

```diff
@@ -24,5 +24,4 @@ fn main() {
     udrop::<A<[u8]>>(A { 0: *foo() });
     //~^ERROR E0277
     udrop::<A<[u8]>>(A(*foo()));
```

Comment:
this needs to get moved to a different test... does this make it all the way to codegen? i'm surprised that the the tuple constructor works.

Reply:
Umm, actually it ICE's,... oops.

Reply:
In general, it's not possible to do "fewer" checks during typeck safely, unless you have a really good reason to believe that it's implied by some other check. The best you can do is something like @cjgillot is talking about, and modify error reporting code to downgrade error messages when they're redundant.

Reply:
Yeah, although *generally* speaking (and with my poor knowledge of the compiler) isn't this check somewhat redundant because you can't do anything with a unsized value. (although not in all cases it seems)